life_expectancy/data_cleaning.py

```diff
@@ -1,53 +1,138 @@
 """Module to clean assignment data"""
+from abc import ABC
 import pandas as pd
-from pandas import DataFrame
+from life_expectancy.loading_saving import DataLoader, CSVDataLoader, JSONDataLoader
+from life_expectancy.country import Country
 
-def clean_data(
-    raw_dataframe: DataFrame,
-    country: str = 'PT'
-):
+class DataCleaner(ABC):
     """
-    This function takes the raw data as a pandas dataframe
-    and returns the cleaned data as a pandas dataframe
+    Data Cleaner class responsible for cleaning several
+    types of data objects
+    """
+    def __init__(self,
+                 file_path : str = None
+                 ) -> None:
+        self.dataloader = DataLoader(file_path)
+
+    def clean_data(self,
+                   data: pd.DataFrame,
+                   country : Country
+                   ) -> pd.DataFrame:
+        """
+        Cleans data
+        Args:
+            country (Country): name of country to filter data
+            data (pd.DataFrame): data object
+        Returns:
+            pd.DataFrame: cleaned final pandas dataframe
+        """
+
+    def load_data(self) -> pd.DataFrame:
+        """Loads the data from file
+        Returns:
+            pd.DataFrame: dataframe with loaded data
+        """
+        return self.dataloader.load_data()
+
+    @staticmethod
+    def check_country_exists(country: str) -> bool:
+        """Test if the country receives is possible
+        Args:
+            country (Country): Country received
+        Returns:
+            bool: True or false based on if it exists
+        """
+        if country.upper() in Country.__members__:
+            return True
+        return False
+
+class CSVDataCleaner(DataCleaner):
+    """Reads data from .csv file and cleans it"""
+
+    def __init__(self,
+                 file_path : str=None
+                 ) -> None:
+
+        super().__init__(file_path)
+        self.loader = CSVDataLoader(file_path)
+
+    def clean_data(self,
+                   data: pd.DataFrame,
+                   country: Country = 'PT'
```

Comment:
I would put this as default value None so the user can choose not to filter

Reply:
Agreed